end_to_end/gpu/a3/test_gemma3_logits.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+# This file is both an integration test that runs once a day on a v4-8 and documentation for how to get started with Gemma3-4b.
+
+# The flow of this file is as follows:
+# 1. Convert the checkpoint downloaded from Kaggle to make it compatible with MaxText
+# 2. Run decoding, finetuning of Gemma3-4b with the converted checkpoint. Also, run pretraining of Gemma3-4b
+# 3. Convert the scanned checkpoint from step 1 into unscanned checkpoint format and run more efficient decoding.
+
+
+set -ex
+idx=$(date +%Y-%m-%d-%H-%M)
+export MODEL_VARIATION='4b'
+export MODEL_NAME=gemma3-${MODEL_VARIATION}
+
+# Installing torch for deps in forward_pass_logit_chekcker.py
+python3 -m pip install torch --index-url https://download.pytorch.org/whl/cpu
+
+# After downloading checkpoints, copy them to GCS bucket at $CHKPT_BUCKET \
+# Non-Googlers please remember to use separate GCS paths for uploading model weights from kaggle ($CHKPT_BUCKET) and MaxText compatible weights ($MODEL_BUCKET).
+# Non-Googlers please remember to point these variables to GCS buckets that you own, this script uses internal buckets for testing.
+# You can use the Flax checkpoint available on Kaggle:
+# https://www.kaggle.com/models/google/gemma-3/flax/
+
+export CHKPT_BUCKET=gs://maxtext-gemma/gemma3/flax
+export MODEL_BUCKET=gs://maxtext-gemma/gemma3
+
+python3 -m MaxText.convert_gemma3_chkpt --base_model_path ${CHKPT_BUCKET}/${MODEL_VARIATION} --maxtext_model_path ${MODEL_BUCKET}/${MODEL_VARIATION}/${idx} --model_size ${MODEL_VARIATION}
+
+# Current MaxText.generate_param_only_checkpoint will need to skip on GPU due to cpu process error. reuse the unscanned ckpt generated seperately.
+
+# # Non-Googlers please remember to point `DATASET_PATH` to the GCS bucket where you have your training data
+# export DATASET_PATH=gs://maxtext-dataset
+# # Non-Googlers please remember to point `BASE_OUTPUT_DIRECTORY` to a GCS bucket that you own, this bucket will store all the files generated by MaxText during a run
+# export BASE_OUTPUT_DIRECTORY=gs://runner-maxtext-logs
+# # We define `CONVERTED_CHECKPOINT` to refer to the checkpoint subdirectory. This way it is easier to use this path in the `train.py` and `decode.py` commands
+# export CONVERTED_CHECKPOINT=${MODEL_BUCKET}/${MODEL_VARIATION}/${idx}/0/items
+# export RUN_NAME=unscanned_chkpt_${idx}
+# export UNSCANNED_CKPT_PATH=${BASE_OUTPUT_DIRECTORY}/${RUN_NAME}/checkpoints/0/items
+# Note that the `CONVERTED_CHECKPOINT` is in a `scanned` format which is great for training but for efficient decoding performance we want the checkpoint in an `unscanned` format.
+# We can do this by running `MaxText/generate_param_only_checkpoint.py` on `CONVERTED_CHECKPOINT` with `force_unroll=true`.
+#JAX_PLATFORMS=cpu python3 -m MaxText.generate_param_only_checkpoint MaxText/configs/base.yml base_output_directory=${BASE_OUTPUT_DIRECTORY} load_parameters_path=${CONVERTED_CHECKPOINT} run_name=${RUN_NAME} model_name=${MODEL_NAME} force_unroll=true
```

Comment:
do you need this?

Reply:
This is the step of how I generate the unscaned ckpt on tpu. 
On gpu, this step generating errors. so I commented those steps out. wdyt?

Reply:
will they generate errors if you run with with `JAX_PLATFORMS=cpu`, also will end_to_end/gpu/a3/test_gemma3_logits.sh be added in nightly tests?

Reply:
yes, it will generate errors as we specify attention=cudnn_flash_te.